### ISSUE TITLE
Who owns https://github.com/haskell?

### ISSUE THREAD

**Issue by tomjaguarpaw** [haskell-org/committee#10]
Prompted by a question by Artem P on Slack it would be helpful to clarify who owns https://github.com/haskell and document that information somewhere.

* Does the Haskell.org committee have overall authority?
* Perhaps some other organisation owns it. If so I can't think which it would be.
* Perhaps it's a loose collective with no formal governance policy. Should we try to come up with something more formal?

[Really this question belongs somewhere under https://github.com/haskell but there seems to be nowhere suitable. (Ideally every GitHub organisation needs a `meta` repository where questions about the organisation itself can be raised.) Since I can't find a proper venue to track this issue, and the committee in principle has an interest, it seems reasonable to track it here.]

**Comment by Bodigrim** [comment]
Excellent question! It seems that historically https://github.com/haskell used to be associated with CLC, but AFAIR it was never properly affiliated. 

There are 14 (!) owners at the moment: https://github.com/orgs/haskell/people?query=role%3Aowner, some of which has not been active for years. Sounds like a huge security vulnerability.

**Comment by gbaz** [comment]
imho the clc should jointly own it with the haskell.org committee and haskell admins.

**Comment by tomjaguarpaw** [comment]
> imho the clc should jointly own it with the haskell.org committee and haskell admins.

Sounds good to me. How should we go about proposing this plan? Post something in Discourse (and Reddit and haskell-cafe)?

**Comment by tomjaguarpaw** [comment]
See also: https://github.com/haskell/core-libraries-committee/issues/43

**Comment by andreasabel** [comment]
@tomjaguarpaw wrote:
> (Ideally every GitHub organisation needs a `meta` repository where questions about the organisation itself can be raised.)

Exactly.

**Comment by Bodigrim** [comment]
> imho the clc should jointly own it with the haskell.org committee and haskell admins.

With all due respect to Haskell.org committee, I do not really see their stake in github.org/haskell. The comittee has a clear [remit](https://www.haskell.org/haskell-org-committee/), (unsurprisingly) limited to haskell.org and its subdomains. 

The rationale behind my opposition is that steering a single committee is challenging enough, but organizing several of them in a joint decision-making body will quickly prove to be next to impossible.

I suggest github.org/haskell to be owned by CLC, as a body which was historically most involved with this organisation and has a clear interest in upkeeping it. Governance changes will require buy in from Hackage admins, embodied in @gbaz, who retain ownership for technical purposes and moderation. This is similar to a recently established management of PVP.

**Comment by gbaz** [comment]
I believe that list is not intended to be exclusive. In particular, during my time on the committee (including as chair) I always viewed the haskell github organization as under its remit as per "hosting and maintaining shared assets and infrastructure for the benefit of the Haskell community" -- and in general, all infrastructure provided to the general haskell community through by third parties, unless otherwise specified, tended to be treated in this way. Github is loose enough with its organizations that it doesn't really matter. But in general, for resources that one specifically asks for, it has made sense to say "here is the body with a status and bank account etc".

I don't mind if the structure is other than that, just that this is what I always figured and operated as what the implicit structure was.

Also given that there are a bunch of non-core libraries in the repo, putting it purely under CLC purview seems a bit odd. That said, I think relatively little governance is necessary. In my experience it has not amounted to more than occasionally accepting reasonable transfers of libraries to this repo (which typically has happen when they are widely used and intended to be put under broader community ownership for more stability) and very rarely handing out sane permissions to trusted individuals, often to help them set up CI integration.

**Comment by Bodigrim** [comment]
FWIW I'm not interested in power games here, I just don't want to participate in joint committees, because I find them mostly impotent. If there arises a community consensus that haskell.org committee can take the burden alone, I would not oppose at all.

**Comment by jaspervdj** [comment]
I think it's a bit fuzzy -- clearly the CLC are the ones who do most of the actual work here.  I occasionally do "admin" sort of stuff, assigning permissions when requested, moving repos and settings.  In the past I've also helped mediate conflicts between parties.  I do not really enjoy doing these things, but I also don't want to the CLC to have to deal with this, since they have much more valuable things to do.

I completely trust everybody on both committees, so I think having e.g. 1-2 owners from CLC, 1-2 owners from Haskell.org, and another owner from the Hackage admins works well in practice to ensure nothing gets blocked.  14 owners sounds crazy!  We should definitely reduce that.

If this does not work out, and we do need one entity to maintain this organization, this should probably be the Haskell Foundation since they are an umbrella organization, and `github.com/haskell` has become a bit of an umbrella thing as well (e.g. HLS is in there, some old stuff, libraries not affiliated with CLC...).  But I'm also very conscious about adding to the HF workload...

To summarize -- I think getting things done is more important than formalizing ownership.  I don't think there are any major decisions that need to be made(?) and if so it would be easy to get consensus.  I just don't want to end up in a situation where one group needs to wait 2-3 days for something to get changed, etc.

**Comment by andreasabel** [comment]
@jaspervdj : A reference is broken in your comment:
> (edit: catching up with #43)

**Comment by jaspervdj** [comment]
We discussed this at the Haskell.org monthly meeting, and we thought a first good step is creating the proposed https://github.com/haskell/meta repository so there is a better place to have these sort of discussions.

**Comment by hasufell** [comment]
> If this does not work out, and we do need one entity to maintain this organization, this should probably be the Haskell Foundation since they are an umbrella organization, and github.com/haskell has become a bit of an umbrella thing as well (e.g. HLS is in there, some old stuff, libraries not affiliated with CLC...). But I'm also very conscious about adding to the HF workload...

I would find it inappropriate if HF manages this organization.

The Haskell Foundation is in my perception a body that fills the gap between industry and community.

Haskell.org/CLC/hackage admins are purely volunteers with none or only very loose connection to industry.

I find it important to keep some separation of concerns, so that the different bodies can focus on different things, which also allows more fruitful disagreements, as well as some level of checks and balances.

**Comment by hasufell** [comment]
> That said, I think relatively little governance is necessary. In my experience it has not amounted to more than occasionally accepting reasonable transfers of libraries to this repo (which typically has happen when they are widely used and intended to be put under broader community ownership for more stability) and very rarely handing out sane permissions to trusted individuals, often to help them set up CI integration

It isn't particularly clear to everyone who should be contacted for different types of issues and it seems no one is really taking over "ownership" and putting that information into a document, visible for people who are running repositories under the haskell organization.

I find these loose affairs unsatisfying and I wish for them to become more concrete, otherwise it's going to put people off of hosting their projects here.

**Comment by tomjaguarpaw** [comment]
Strongly agree with @hasufell. Without a clear statement somewhere of the current authorities and processes (even if those processes are only implicitly defined) it can waste a huge amount of people's time just working out who to contact and how.

That said, I suggest we carry on at [the issue opened on the Haskell organisation meta repo for this purpose](https://github.com/haskell/meta/issues/1) and close this discussion.

**Comment by tomjaguarpaw** [comment]
Closing, as the discussion has moved to https://github.com/haskell/meta/issues/1
